pkg/operator/ceph/config/conditions.go

```diff
@@ -47,23 +41,19 @@ func ConditionExport(context *clusterd.Context, namespaceName types.NamespacedNa
 
 // setCondition updates the conditions of the cluster custom resource
 func setCondition(c *clusterd.Context, namespaceName types.NamespacedName, newCondition cephv1.Condition) {
-	ctx := context.TODO()
-	cluster, err := c.RookClientset.CephV1().CephClusters(namespaceName.Namespace).Get(ctx, namespaceName.Name, metav1.GetOptions{})
+	cluster, err := getCluster(c, namespaceName)
 	if err != nil {
-		if kerrors.IsNotFound(err) {
-			logger.Errorf("no CephCluster could not be found. %+v", err)
-			return
-		}
-		logger.Errorf("failed to get CephCluster object. %+v", err)
 		return
 	}
-
-	if conditions == nil {
-		conditions = &cluster.Status.Conditions
-		if cluster.Status.Conditions != nil {
-			conditionMapping(*conditions)
+	conditionMap := conditionMapping(cluster.Status.Conditions)
+	if newCondition.Type == cephv1.ConditionReady {
+		checkConditionFalse(c, namespaceName, conditionMap)
```

Comment:
This code would really be improved if we computed all the conditions at once and had a single update to the CephCluster CR. Currently it involves multiple updates. But this pattern already existed, so we can keep it for now instead of blocking this PR.